### ISSUE TITLE
ignoreMethods has now effect 

### ISSUE THREAD

**Issue by GM-Alex** [woutervh-/typescript-is#27]
Using ignoreMethods has no effect for the following data

```
const test = {
    list: {
        listItem: "aaa",
        data: {
            id: {
                some: "span.to-details",
                convert: x => x
            }
        }
    }
};

```

validate against the following interface

```
interface IElement {
    some?: string;
    convert?: (value: string) => string;
}

interface IMap {
    id: string|IElement;
}

interface ITest {
    list: {
        listItem: string;
        data: IMap;
        convert?: (value: string) => string;
    };
}
```

by using `assertEquals<ITest>(test)`. This results in `TypeGuardError: validation failed at $.list.data.id: there are no valid alternatives`. The following data results in `TypeGuardError: validation failed at $.list.convert: expected an object`

```
const test = {
    list: {
        listItem: "aaa",
        data: {
            id: {
                some: "span.to-details"
            }
        },
        convert: x => x
    },
};
```

Using

```
const test = {
    list: {
        listItem: "aaa",
        data: {
            id: {
                some: "span.to-details"
            }
        }
    },
};
```

works as expected.

**Comment by woutervh-** [comment]
Hi @GM-Alex 

The correct syntax for methods is the following:

```typescript
const test = {
  convert(x) {
    return x;
  }
};

interface ITest {
  convert(value: string): string;
}
```

The syntax you have used declares an object property where the value is a lambda function.
However I see your point and it makes sense to also ignore functions.
I will work on a fix for this when I get some time to work on it.

**Comment by GM-Alex** [comment]
Hi @woutervh-,

thanks for your feedback. I made a PR that should also include lambda functions if the ignore flag is set.

**Comment by woutervh-** [comment]
Thanks for the PR @GM-Alex 🎉

I will check it out when I get home.

**Comment by woutervh-** [comment]
Released version `0.12.1`:
https://github.com/woutervh-/typescript-is/releases/tag/v0.12.1

**Comment by woutervh-** [comment]
FYI in the latest version `ignoreMethods` will really only ignore methods.
To ignore functions, even in objects, use the new `ignoreFunctions`.

https://github.com/woutervh-/typescript-is/releases/tag/v0.14.3
